### PR TITLE
[#IA-7] Tests for report about medication consumption

### PR DESCRIPTION
**Pull request by dragana-todorovic**

### CODE REVIEW

File: IntegrationAdapters/Controllers/SftpController.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public SftpController()
         public IActionResult Post(Interval interval)
         {
             List<MedicineReport> result = new List<MedicineReport>();
-            result = medicineReportService.GetByDateInterval(new TimeInterval(DateTime.Parse(interval.start), DateTime.Parse(interval.end)));
+            result = medicineReportService.GetByDateInterval(medicineReportService.GetAll(), new TimeInterval(DateTime.Parse(interval.start), DateTime.Parse(interval.end)));
```
Comment (milicasiriski) [inline IntegrationAdapters/Controllers/SftpController.cs]: no need to reassign result, you could just assign this in the first line

File: IntegrationAdapters/Services/MedicineReportService.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 namespace IntegrationAdapters.Services
 {
-    public class MedicineReportService
+    public class MedicineReportService : IMedicineReportRepository
```
Comment (milicasiriski) [inline IntegrationAdapters/Services/MedicineReportService.cs]: Why does service implement repository interface?

File: IntegrationAdaptersTests/RegisterHospitalTests.cs [hunk 3]
```diff
@@ -21,20 +21,20 @@ public void Is_pharmacy_already_existing_on_hospital()
             ApiService service = new ApiService();
 
             Api api = new Api("k", "k", "k");
-            bool result = service.RegisterHospitalOnPharmacy(api);
+            bool result = service.IsPharmacyExistsOnHospital(api);
```
Comment (milicasiriski) [inline IntegrationAdaptersTests/RegisterHospitalTests.cs:24]: 💯 Are these the tests that failed if you run them again? 😄 This should work

File: IntegrationAdaptersTests/SftpTests.cs [hunk 4]
```diff
@@ -9,12 +12,37 @@ namespace IntegrationAdaptersTests
 {
     public class SftpTests
     {
+        public Mock<IMedicineReportRepository> mock;
         [Fact]
         public void sendFileSuccess()
```
Comment (milicasiriski) [inline IntegrationAdaptersTests/SftpTests.cs]: naming convention

File: IntegrationAdaptersTests/SftpTests.cs [hunk 5]
```diff
@@ -9,12 +12,37 @@ namespace IntegrationAdaptersTests
...
             SftpService service = new SftpService();
             bool success = service.SendFile(@"C:\Users\dragana\Desktop\Projekat PSW\Healthcare-System\IntegrationAdaptersTests\bin\Debug\netcoreapp3.1\testFile.txt");
             Assert.True(success);   
         }
+
+        [Fact]
+        public void Find_existing_report()
+        {
+            mock = new Mock<IMedicineReportRepository>();
+            mock.Setup(expression: t => t.GetAll()).Returns(new List<MedicineReport> { new MedicineReport { Id = "1", Date = new DateTime(2020, 10, 5), Dosage = new List<MedicineDosage>() }, new MedicineReport { Id = "2", Date = new DateTime(2020, 5, 10), Dosage = new List<MedicineDosage>() } });
```
Comment (milicasiriski) [inline IntegrationAdaptersTests/SftpTests.cs]: If you're using objects for stubs, move them to TestConstants class in util package or make fields in this class (if they're used only in this test)

**Review by milicasiriski — CHANGES_REQUESTED** [review]
Service shouldn't implement IRepository and minor fixups needed

File: IntegrationAdapters/Services/MedicineReportService.cs (hunk 2 above)
Comment (dragana-todorovic) [reply IntegrationAdapters/Services/MedicineReportService.cs]: Ups, lapsus.. 
It was supposed to be a MedicineReportRepository :(

File: IntegrationAdaptersTests/RegisterHospitalTests.cs (hunk 3 above)
Comment (dragana-todorovic) [reply IntegrationAdaptersTests/RegisterHospitalTests.cs:24]: Yes, but now it is pass every time.
Thanks for suggestion 😄

File: IntegrationAdapters/Services/MedicineReportService.cs (hunk 2 above)
Comment (milicasiriski) [reply IntegrationAdapters/Services/MedicineReportService.cs]: It's ok hahaha you know what to fix then 😄

File: IntegrationAdapters/Services/MedicineReportService.cs [hunk 6]
```diff
@@ -27,11 +27,12 @@ public List<MedicineReport> GetAll()
             return medicineReportRepository.GetAll();
         }
 
-        public List<MedicineReport> GetByDateInterval(TimeInterval timeInterval)
+        public List<MedicineReport> GetByDateInterval(List<MedicineReport> reports, TimeInterval timeInterval)
```
Comment (milicasiriski) [inline IntegrationAdapters/Services/MedicineReportService.cs]: I think the reports parameter is useless

File: IntegrationAdapters/Services/MedicineReportService.cs [hunk 7]
```diff
@@ -27,11 +27,12 @@ public List<MedicineReport> GetAll()
             return medicineReportRepository.GetAll();
         }
 
-        public List<MedicineReport> GetByDateInterval(TimeInterval timeInterval)
+        public List<MedicineReport> GetByDateInterval(List<MedicineReport> reports, TimeInterval timeInterval)
         {
             List<MedicineReport> result = new List<MedicineReport>();
+            reports = GetAll();
```
Comment (milicasiriski) [inline IntegrationAdapters/Services/MedicineReportService.cs]: since you just get them all here

File: IntegrationAdaptersTests/SftpTests.cs (hunk 4 above)
Comment (milicasiriski) [reply IntegrationAdaptersTests/SftpTests.cs]: just rename this method to Sends_files_successfully

**Review by milicasiriski — APPROVED** [review]
Correct those 2 more things and then you can merge 😄

**Review by lukaDoric — APPROVED** [review]

**Pull request merged**